src/main.rs

```diff
@@ -1,31 +1,68 @@
-use std::{
-    path::{Path, PathBuf},
-    time::Instant,
-};
+use std::{path::PathBuf, time::Instant};
 
+use clap::Parser;
+use file::read_3d_image;
 use glam::{uvec2, UVec2};
+use ndarray::{s, Array2, ShapeBuilder};
 
 mod file;
 mod graphics;
 
 type Image = image::RgbaImage;
+type ImageSlice = Array2<u8>;
```

Comment:
Great stuff. I was hoping we could use the data directly without relying on the `image` crate.